tests/functional/cURLTest.php

```diff
@@ -51,20 +51,55 @@ public function queryRequestBody()
 		$this->assertEquals('foo=bar', $r->body);
 	}
 
+	/** @test */
+	public function queryRequestEmptyArrayBody()
+	{
+		$r = $this->makeCurl()->post(static::URL.'/echo.php', array());
+		$this->assertEquals('', $r->body);
```

Comment:
Maybe check the `content-length` header in all these new tests, to check for the bug you originally wanted to fix?

Reply:
Just the fact that the request is executed successfully without any errors or exceptions means that content-length was correct, any more checks would be redundant.
In fact, you can see that a few of these tests fail if you revert the changes in the source code and try to run them, to assert that they do cover this case.

Reply:
Ah, okay, good to know. I don't have php installed locally anymore so I don't have the option to check myself :o